### PR TITLE
WIP: Memory leak in networking code

### PR DESCRIPTION
**Pull request by coot**

### CODE REVIEW

**Comment by coot** [comment]
Requires: https://github.com/input-output-hk/cardano-base/pull/89
Is requirement for: https://github.com/input-output-hk/cardano-node/pull/762

File: network-mux/src/Network/Mux/Bearer/Socket.hs [hunk 1]
```diff
@@ -72,17 +73,17 @@ socketAsMuxBearer sduTimeout_m tracer sd =
                 Nothing -> do
                     traceWith tracer $ Mx.MuxTraceSDUReadTimeoutException
                     throwM $ Mx.MuxError Mx.MuxSDUReadTimeout "Mux SDU Timeout" callStack
-                Just r -> return r
+                Just !r -> return r
```
[the remark this answers is not in the document]
Comment (dcoutts) [reply network-mux/src/Network/Mux/Bearer/Socket.hs]: Pushed a patch to implement this suggestion.

**Comment by dcoutts** [comment]
> Plus we can actually avoid changing cardano-base.

**Comment by dcoutts** [comment]
bors merge

**Pull request merged**